Source/CombatExtended/CombatExtended/Verbs/Verb_MeleeAttackCE.cs

```diff
@@ -261,10 +270,104 @@ private IEnumerable<DamageInfo> DamageInfosToApply(LocalTargetInfo target, bool
+                    var tempDInfo = damageInfo;
+                    bool armorDeflected;
+                    bool armorReduced;
+                    bool shieldAbsorbed;
+
+                    var dmgList = new List<BodyPartRecord>();
+                    var noDmgList =  new List<BodyPartRecord>();
+
+                    // I calculate the added hit chance for vulnerable body parts
+                    float hitMult = meleeSkill / 20.0f * MeleeSkillChanceMult + 1.0f;
+
+                    // For each bodypart the target has
+                    foreach (BodyPartRecord part in pawnParts)
+                    {
+                        // If it can be targeted
+                        if (part.def.conceptual)
+                            continue;
+
+                        // Check how much damage would be dealt when attacking it
+                        LastAttackVerb = this;
+                        tempDInfo = ArmorUtilityCE.GetAfterArmorDamage(damageInfo, p, part, out armorDeflected, out armorReduced, out shieldAbsorbed, true);
+                        LastAttackVerb = null;
+
+                        // Store part into list, based on whether damage would be dealt or not
+                        if(tempDInfo.Amount > 0)
+                        {
+                            dmgList.Insert(Rand.RangeInclusive(0,dmgList.Count), part); // Insert in random order
+                        }
+                        else
+                        {
+                            noDmgList.Add(part);
+                        }
+                            
+                    }
+
+                    // If there is at least 1 vulnerable body part
+                    if (dmgList.Count > 0)
+                    {
+                        float seed = Rand.Value;
+                        bool selected = false;
+
+                        // Check if the pawn should strike a vulnerable bodypart, taking into account coverage
+                        foreach (BodyPartRecord b in dmgList)
+                        {
+                            if(seed < b.coverageAbs * hitMult)
+                            {
+                                damageInfo.SetHitPart(b);
+                                selected = true;
+                                break;
+                            }
+                            else
+                            {
+                                seed -= b.coverageAbs * hitMult;
+                            }
+                            
+                        }
+
+                        // If the check fails, target anything else, coverage doesn't matter here as damage dealt will be 0
+                        if (!selected)
+                        {
+                            damageInfo.SetHitPart(noDmgList[Rand.Range(0, noDmgList.Count)]); // doesn't matter which, always deals 0 damage, ignoring coverage
```

Comment:
This code block should probably be removed. If a suitable body part isn't found, it's better to let the game target the body part it originally set, rather than overwrite it with a random dice roll.

Reply:
The idea behind this line is that not doing so will let the game roll again for one of the vulnerable parts.
So if my pawn has 44% chance to hit the head and misses, removing this line leads to the chance rising to 48%

If you are ok with that i can remove the line.